tmt/base.py

```diff
@@ -2190,49 +2479,58 @@ def print_header(self):
         header += 'id'
         echo(style(header, fg='blue'))
 
-    def show(self):
+    def show(self) -> None:
         """ Display the current status """
         # Prepare absolute workdir path if --id was used
         id_ = self.opt('id')
         root_path = self.opt('workdir-root')
         self.print_header()
+        assert self._context_object is not None  # narrow type
+        assert self._context_object.tree is not None  # narrow type
         for abs_path in tmt.utils.generate_runs(root_path, id_):
-            run = Run(abs_path, self._context.obj.tree, self._context)
+            run = Run(id_=abs_path, tree=self._context_object.tree, context=self._context)
             self.process_run(run)
 
 
 class Clean(tmt.utils.Common):
     """ A class for cleaning up workdirs, guests or images """
 
-    def __init__(self, parent=None, name=None, workdir=None, context=None):
+    def __init__(self,
+                 parent: Optional[tmt.utils.Common] = None,
+                 name: Optional[str] = None,
+                 workdir: tmt.utils.WorkdirArgumentType = None,
+                 context: Optional[click.Context] = None) -> None:
         """
         Initialize name and relation with the parent object
 
         Always skip to initialize the work tree.
         """
         # Set the option to skip to initialize the work tree
-        context.params[tmt.utils.PLAN_SKIP_WORKTREE_INIT] = True
+        if context:
+            context.params[tmt.utils.PLAN_SKIP_WORKTREE_INIT] = True
         super().__init__(parent, name, workdir, context)
 
-    def images(self):
+    def images(self) -> bool:
         """ Clean images of provision plugins """
         self.info('images', color='blue')
         successful = True
         for method in tmt.steps.provision.ProvisionPlugin.methods():
-            if not method.class_.clean_images(self, self.opt('dry')):
+            # TODO: when generic, there would be no doubth about whether `clean_images`
```

Comment:
```suggestion
            # TODO: when generic, there would be no doubt about whether `clean_images`
```

Reply:
Fixed in 3017568a8eed9792f95bd9528d2c0a22cedd84a6.